cvat/apps/engine/task.py

```diff
@@ -1140,6 +1135,18 @@ def update_status(msg: str) -> None:
             else:
                 manifest.init_index()
 
+            # Align manifest order with the selected sorting method for image-based tasks.
+            try:
+                if data['sorting_method'] != models.SortingMethod.PREDEFINED:
+                    desired_order = [
+                        os.path.relpath(extractor.get_path(frame_id), upload_dir)
+                        for frame_id in extractor.frame_range
+                    ]
+                    manifest.reorder(desired_order)
+            except Exception:
+                # If reordering fails for some reason, proceed without altering
+                raise ValidationError("Manifest reordering failed")
```

Comment:
```suggestion
            if data['sorting_method'] != models.SortingMethod.PREDEFINED:
                desired_order = [
                    os.path.relpath(extractor.get_path(frame_id), upload_dir)
                    for frame_id in extractor.frame_range
                ]
                try:
                    manifest.reorder(desired_order)
                except Exception:
                    # If reordering fails for some reason, proceed without altering
                    raise ValidationError("Manifest reordering failed")
```
Please check if the exception class can be narrowed. It you want to catch a broad exception class, minimize the scope that can throw an exception.